core/src/main/java/org/apache/accumulo/core/util/threads/ThreadPools.java

```diff
@@ -99,9 +100,14 @@ public static final ThreadPools getServerThreadPools() {
     return SERVER_INSTANCE;
   }
 
-  public static final ThreadPools getClientThreadPools(UncaughtExceptionHandler ueh) {
+  public static final ThreadPools getClientThreadPools(AccumuloConfiguration conf,
+      UncaughtExceptionHandler ueh) {
     ThreadPools clientPools = new ThreadPools(ueh);
-    clientPools.setMeterRegistry(Metrics.globalRegistry);
+    if (conf.getBoolean(Property.GENERAL_MICROMETER_ENABLED) == false) {
+      clientPools.disableThreadPoolMetrics();
+    } else {
+      clientPools.setMeterRegistry(Metrics.globalRegistry);
+    }
```

Comment:
To improve readability, this could be written as:

```java
    if (conf.getBoolean(Property.GENERAL_MICROMETER_ENABLED)) {
      clientPools.setMeterRegistry(Metrics.globalRegistry);
    } else {
      clientPools.disableThreadPoolMetrics();
    }
```